dashboards-reports/public/components/report_definitions/delivery/delivery.tsx

```diff
@@ -55,25 +66,47 @@ export function ReportDelivery(props: ReportDeliveryProps) {
     editDefinitionId,
     reportDefinitionRequest,
     httpClientProps,
+    showDeliveryChannelError,
+    deliveryChannelError
   } = props;
 
-  const [emailCheckbox, setEmailCheckbox] = useState(false);
+  const [sendNotification, setSendNotification] = useState(false);
+  const [channels, setChannels] = useState([]);
+  const [selectedChannels, setSelectedChannels] = useState([]);
+  const [notificationSubject, setNotificationSubject] = useState('');
+  const [notificationMessage, setNotificationMessage] = useState('');
+  const [selectedTab, setSelectedTab] = React.useState<'write' | 'preview'>(
+    'write'
+  );
+  const [testMessageConfirmation, setTestMessageConfirmation] = useState('');
+
+  const handleSendNotification = (e: { target: { checked: boolean | ((prevState: boolean) => boolean); }; }) => {
```

Comment:
could you double check the type? i don't think `e.target.checked` is ever a function?